src/benchmarklib/abstract_table_generator.cpp

```diff
@@ -0,0 +1,60 @@
+#include "abstract_table_generator.hpp"
+
+#include "benchmark_config.hpp"
+#include "benchmark_table_encoder.hpp"
+#include "operators/export_binary.hpp"
+#include "storage/storage_manager.hpp"
+
+namespace opossum {
+
+AbstractTableGenerator::AbstractTableGenerator(const std::shared_ptr<BenchmarkConfig>& benchmark_config)
+    : _benchmark_config(benchmark_config) {}
+
+void AbstractTableGenerator::generate_and_store() {
+  auto table_info_by_name = generate();
+
+  /**
+   * Encode the Tables
+   */
+  for (auto& [table_name, table_info] : table_info_by_name) {
+    table_info.re_encoded = BenchmarkTableEncoder::encode(table_name, table_info.table,
+                                                          _benchmark_config->encoding_config, _benchmark_config->out);
+  }
+
+  /**
+   * Write the Tables into binary files if required
+   */
+  if (_benchmark_config->cache_binary_tables) {
+    for (auto& [table_name, table_info] : table_info_by_name) {
+      if (!table_info.loaded_from_binary || table_info.re_encoded) {
+        auto binary_file_path = std::filesystem::path{};
+        if (table_info.binary_file_path) {
```

Comment:
Who would set this?

Reply:
The derived TableGenerator, if he so sees fit. The FileBasedTableGenerator currently does so if the table already exists as a binary. The TPCHTableGenerator can do so as well, once we figured out where and how and if we want to cache generated TPCH tables.